### PR TITLE
[AKS] `az aks create/update`: Add `--nrg-lockdown-restriction-level` parameter to support specifying restriction level on the managed node resource group

### PR DESCRIPTION
**Pull request by y1hao**
**Related command**

`az aks create`, `az aks update`

**Description**

This PR introduces the `--nrg-lockdown-restriction-level` argument to allow configuring this value. Allowed values are `ReadOnly` and `Unrestricted`. Public doc: https://learn.microsoft.com/en-us/azure/aks/node-resource-group-lockdown

This argument is currently available as extension. This PR ports it to CLI, without any behaviour changes. See: https://github.com/Azure/azure-cli-extensions/pull/5751

**Testing Guide**

Creating a cluster with NRG lockdown enabled:

```sh
$az aks create -n mycluster -g mygroup --nrg-lockdown-restriction-level ReadOnly --aks-custom-headers AKSHTTPCustomFeatures=Microsoft.ContainerService/NRGLockdownPreview
```

Disable NRG lockdown for a cluster:

```sh
$az aks update -n mycluster -g mygroup --nrg-lockdown-restriction-level Unrestricted --aks-custom-headers AKSHTTPCustomFeatures=Microsoft.ContainerService/NRGLockdownPreview
```

**History Notes**

[AKS] `az aks create/update`: Add `--nrg-lockdown-restriction-level` parameter to support specifying restriction level on the managed node resource group

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -8421,6 +8421,44 @@ def test_aks_run_command(self, resource_group, resource_group_location):
...
+            self.check('nodeResourceGroupProfile.restrictionLevel', 'ReadOnly')
+        ])
+
+        # update: ReadOnly -> Unrestricted
+        self.cmd('aks update --resource-group={resource_group} --name={name} --nrg-lockdown-restriction-level Unrestricted --aks-custom-headers AKSHTTPCustomFeatures=Microsoft.ContainerService/NRGLockdownPreview', checks=[
+            self.check('provisioningState', 'Succeeded'),
+            self.check('nodeResourceGroupProfile.restrictionLevel', 'Unrestricted')
+        ])
+
+        # update: Unrestricted -> ReadOnly
```
Comment (y1hao) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:8452]: Added one more test case compared with the tests in extension, to provide more coverage.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 2]
```diff
@@ -8421,6 +8421,44 @@ def test_aks_run_command(self, resource_group, resource_group_location):
...
+            'location': resource_group_location,
+            'ssh_key_value': self.generate_ssh_keys()
+        })
+
+        # create
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} --location={location} ' \
+                     '--enable-managed-identity ' \
+                     '--nrg-lockdown-restriction-level ReadOnly ' \
+                     '--ssh-key-value={ssh_key_value} ' \
+                     '--aks-custom-headers AKSHTTPCustomFeatures=Microsoft.ContainerService/NRGLockdownPreview'
```
Comment (y1hao) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:8440]: We still need this until we clean up the implementation to remove the dependency on the custom feature.

**Review by FumingZhang — COMMENTED** [review]
lgtm

Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=108695072&view=results) to validate the change.

- test_aks_create_and_update_with_nrg_restriction_level

**Comment by y1hao** [comment]
Hi, @yanzhudd, @FumingZhang The previous test run discovered there were failed test cases in the 2020-09-01-hybrid profile. I believe I've fixed them and verified it locally. Could you help rerun a full test please? (Not sure if I could just trigger it myself) Thanks~

**Comment by wangzelin007** [comment]
Hi @y1hao 
Can you merge the latest code from dev to your branch?

**Comment by y1hao** [comment]
@wangzelin007 Thanks. Merged latest dev.

**Review by FumingZhang — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**